### PR TITLE
Add instructions to download the entire dataset

### LINKED ISSUE #4971: Write a piece of documentation on data.download_all() in the install procedure
## Description

In https://scikit-image.org/docs/stable/install.html
write about https://scikit-image.org/docs/stable/api/skimage.data.html#download-all

### PR DESCRIPTION
**Pull request by sciunto**
Closes #4971

### CODE REVIEW

File: INSTALL.rst [hunk 1]
```diff
@@ -161,6 +161,20 @@ or other Python packages is not your best option:
...
+
+Part of our dataset used in our examples is hosted online and is not installed
+by default by the procedures explained above. Data are downloaded once, at the
+first call, but this requires an internet connection. If you prefer to download
+the entire dataset to be able to work offline, you can run this command:
+
+.. code-block:: sh
+
+    python -c 'from skimage.data import download_all; download_all()'
+
```
Comment (emmanuelle) [inline INSTALL.rst:176]:
```suggestion
or call `download_all()` in your favourite interactive Python environment (IPython, Jupyter notebook, ...)
```

**Comment by emmanuelle** [comment]
Thanks @sciunto ! Travis seems to be broken (again) but this is probably not related to your PR.

**Comment by sciunto** [comment]
i merged your suggestion.

File: INSTALL.rst [hunk 2]
```diff
@@ -161,6 +161,20 @@ or other Python packages is not your best option:
...
+Part of our dataset used in our examples is hosted online and is not installed
+by default by the procedures explained above. Data are downloaded once, at the
+first call, but this requires an internet connection. If you prefer to download
+the entire dataset to be able to work offline, you can run this command:
+
+.. code-block:: sh
+
+    python -c 'from skimage.data import download_all; download_all()'
+or call `download_all()` in your favourite interactive Python environment (IPython, Jupyter notebook, ...).
```
Comment (emmanuelle) [inline INSTALL.rst:176]:
```suggestion

or call `download_all()` in your favourite interactive Python environment 
(IPython, Jupyter notebook, ...).
```

**Comment by emmanuelle** [comment]
My suggestion had introduced a formatting pb, I merged a fix to your branch. Will merge the PR if CI is green.

File: INSTALL.rst [hunk 3]
```diff
@@ -161,6 +161,22 @@ or other Python packages is not your best option:
   make system-wide Python changes (installing as root or using sudo),
   you can break the operating system.
 
+
+Downloading the entire dataset
+------------------------------------------------------------------------------
+
+Part of our dataset used in our examples is hosted online and is not installed
```
Comment (grlee77) [inline INSTALL.rst:168]:
```suggestion
Some of the data used in our examples is hosted online and is not installed
```
Nitpick: this minor change sounds better to me

File: INSTALL.rst [hunk 4]
```diff
@@ -161,6 +161,22 @@ or other Python packages is not your best option:
   make system-wide Python changes (installing as root or using sudo),
   you can break the operating system.
 
+
+Downloading the entire dataset
```
Comment (jni) [inline INSTALL.rst:165]: "the entire dataset" is a bit unclear without context. How about "Downloading all demo datasets"?

**Review by jni — APPROVED** [review]
Other than my nitpicky suggestion, I approve. =)

File: INSTALL.rst (hunk 4 above)
Comment (emmanuelle) [inline INSTALL.rst:165]:
```suggestion
Downloading all demo datasets
```

Comment (emmanuelle) [reply INSTALL.rst:165]: ping @sciunto if you accept this suggestion we can merge the PR :-)

Comment (sciunto) [reply INSTALL.rst:165]: I do!

**Pull request merged**